src/Infrastructure.VS/SolutionInfoProvider.cs

```diff
@@ -0,0 +1,61 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using SonarLint.VisualStudio.Core;
+
+namespace SonarLint.VisualStudio.Infrastructure.VS
+{
+    [Export(typeof(ISolutionInfoProvider))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class SolutionInfoProvider : ISolutionInfoProvider
+    {
+        private readonly IServiceProvider serviceProvider;
+        private readonly IThreadHandling threadHandling;
+
+        [ImportingConstructor]
+        public SolutionInfoProvider([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider,
+            IThreadHandling threadHandling)
+        {
+            this.serviceProvider = serviceProvider;
+            this.threadHandling = threadHandling;
+        }
+
+        public async Task<string> GetSolutionNameAsync()
```

Comment:
We need the current solution name to save the Roslyn suppressions file in the correct place. This PR adds an abstraction to fetch it.

We have very similar code in several other places that uses `IVsSolution` e.g. [MigrationSettingsProvider](https://github.com/SonarSource/sonarlint-visualstudio/blob/5cd8033fbb2cde08c631cfaaf5851fabcfa11f16/src/ConnectedMode/Migration/MigrationSettingsProvider.cs#L40).

The longer-term aim would be to refactor the existing code to use this abstraction. However, it's out of scope for this release, so I've created a separate ticket for it.

In fact, I've created two: one to refactor existing code to use this method (#4481), and a second to extend the abstraction to return the directory name too (#4482), as that code is duplicated in several places too.